include/crow/utility.h

```diff
@@ -237,6 +237,46 @@ namespace crow
             static constexpr bool value = sizeof(__test<F, Args...>(0)) == sizeof(char);
         };
 
+        // Check Tuple contains type T
+        template<typename T, typename Tuple>
+        struct has_type;
+
+        template<typename T>
+        struct has_type<T, std::tuple<>> : std::false_type
+        {};
+
+        template<typename T, typename U, typename... Ts>
+        struct has_type<T, std::tuple<U, Ts...>> : has_type<T, std::tuple<Ts...>>
+        {};
+
+        template<typename T, typename... Ts>
+        struct has_type<T, std::tuple<T, Ts...>> : std::true_type
+        {};
+
+        // Check F is callable with Args
+        template<typename F, typename... Args>
+        struct is_callable
+        {
+            template<typename F2, typename... Args2>
+            static std::true_type __test(decltype(std::declval<F2>()(std::declval<Args2>()...))*)
+            {
+                return {};
+            }
+
+            template<typename F2, typename... Args2>
+            static std::false_type __test(...)
+            {
+                return {};
+            }
```

Comment:
I think it will be better to remove function definitions here cause if u use TRUE metaprogramming u doesn't care about function bodies and similar. Only return types.